fe/src/main/java/org/apache/doris/qe/QueryDetailQueue.java

```diff
@@ -0,0 +1,58 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.qe;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.util.List;
+import java.util.Map;
+import org.apache.doris.qe.QueryDetail;
+
+// Queue of QueryDetail.
+// It's used to collect queries for monitor.
+// The default copacity is 10000.
+public class QueryDetailQueue {
+    private static Map<String, QueryDetail> queryDetails = Maps.newHashMap();
+    private static int queryNumber = 0;
+    private static int queryCapacity = 10000;
+
+    public static synchronized void addOrUpdateQueryDetail(QueryDetail queryDetail) {
+        if (queryDetails.get(queryDetail.getQueryId()) == null) {
```

Comment:
You could use putIfAbsent

Reply:
Because I need to increment queryNumber according to the judgement.